### PR TITLE
ATO-446: Set up alerting

### PR DESCRIPTION
**Pull request by ethanmills**
## What
- Setup lambda subcribed to sns which posts to slack
- Update all metrics to be correct
- VPC changes required as some functions must not be in VPC for egress, and previous configuration did not work

### CODE REVIEW

File: orchestration-alerting/slackNotifications.js [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        colorCode,
+        snsMessageFooter
+    );
+
+    console.log("Sending alert to slack");
+    try {
+        // eslint-disable-next-line no-undef
+        const response = await fetch(slackHookUrl, messageRequest);
+        const message = await response.text();
+        console.log(message);
```
[the remark this answers is not in the document]
Comment (ethanmills) [reply orchestration-alerting/slackNotifications.js:117]: Yeah, this is the only information you get about failure responses, it's useful to have always

File: template.yaml [hunk 2]
```diff
@@ -392,7 +391,7 @@ Resources:
               MetricName: Invocations
               Dimensions:
                 - Name: FunctionName
-                  Value: !Ref OpenIdConfigurationFunction.Alias
+                  Value: !Sub ${Environment}-OpenIdConfigurationFunction
```
[the remark this answers is not in the document]
Comment (ethanmills) [reply template.yaml:394]: Yeah, we need the function name not the alias name. This is not available

File: orchestration-alerting/slackNotifications.js [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+const buildMessageRequest = async function (snsMessage, colorCode, snsMessageFooter) {
+    const body = formatMessage(snsMessage, colorCode, snsMessageFooter);
+    if (process.env.DEPLOY_ENVIRONMENT === "integration") {
+        body.channel = process.env.SLACK_CHANNEL_ID ||
+            (await getParameter(process.env.DEPLOY_ENVIRONMENT + "-slack-channel-id"));
+    }
+    if (process.env.DEPLOY_ENVIRONMENT === "dev" || process.env.DEPLOY_ENVIRONMENT === "staging") {
+        body.channel = process.env.SLACK_CHANNEL_ID ||
+            (await getParameter(process.env.DEPLOY_ENVIRONMENT + "-slack-channel-id"));
+    }
```
[the remark this answers is not in the document]
Comment (ethanmills) [reply orchestration-alerting/slackNotifications.js]: see https://github.com/govuk-one-login/authentication-smoke-tests/pull/507#discussion_r1601247269

[the remark this answers is not in the document]
Comment (ethanmills) [reply orchestration-alerting/slackNotifications.js]: This integration defaults to sending to 2nd line

[the remark this answers is not in the document]
Comment (CarlyG55) [reply orchestration-alerting/slackNotifications.js]: Whoops I forgot I asked this before haha

**Review by CarlyG55 — APPROVED** [review]

**Pull request merged**